File: Density/app/src/main/java/org/cornelldti/density/density/util/ValueFormatter.java
package org.cornelldti.density.density.util;

import com.github.mikephil.charting.data.Entry;
import com.github.mikephil.charting.formatter.IValueFormatter;
import com.github.mikephil.charting.utils.ViewPortHandler;

import java.text.DecimalFormat;

public class ValueFormatter implements IValueFormatter {

    private DecimalFormat format;

    public ValueFormatter() {
        format = new DecimalFormat("#.###");
    }

    @Override
    public String getFormattedValue(float value, Entry entry, int dataSetIndex, ViewPortHandler viewPortHandler) {
        return format.format(value);
    }
}

File: Density/app/src/main/java/org/cornelldti/density/density/util/ValueFormatter.kt
package org.cornelldti.density.density.util

import com.github.mikephil.charting.data.Entry
import com.github.mikephil.charting.formatter.IValueFormatter
import com.github.mikephil.charting.utils.ViewPortHandler

import java.text.DecimalFormat

class ValueFormatter : IValueFormatter {

    private val format: DecimalFormat = DecimalFormat("#.###")

    override fun getFormattedValue(value: Float, entry: Entry, dataSetIndex: Int, viewPortHandler: ViewPortHandler): String {
        return format.format(value.toDouble())
    }
}
